beacon-chain/core/light-client/lightclient.go

```diff
@@ -551,3 +552,193 @@ func NewLightClientUpdateFromOptimisticUpdate(update *ethpbv2.LightClientOptimis
 		SignatureSlot:  update.SignatureSlot,
 	}
 }
+
+func ComputeTransactionsRoot(payload interfaces.ExecutionData) ([]byte, error) {
+	transactionsRoot, err := payload.TransactionsRoot()
+	if errors.Is(err, consensus_types.ErrUnsupportedField) {
+		transactions, err := payload.Transactions()
+		if err != nil {
+			return nil, errors.Wrap(err, "could not get transactions")
+		}
+		transactionsRootArray, err := ssz.TransactionsRoot(transactions)
+		if err != nil {
+			return nil, errors.Wrap(err, "could not get transactions root")
+		}
+		transactionsRoot = transactionsRootArray[:]
+	} else if err != nil {
+		return nil, errors.Wrap(err, "could not get transactions root")
+	}
+	return transactionsRoot, nil
+}
+
+func ComputeWithdrawalsRoot(payload interfaces.ExecutionData) ([]byte, error) {
```

Comment:
No need to export

Reply:
> No need to export

not exporting leads to function not found in lightclient_test.go